File: text_generator/local_text_generator.py
chars = ["0","1","2","3","4","5","6","7","8","9","a","b","c","d","e","f","g","h","i"]#what characters code can use and in what order or idunno
#^^^ alias bytes﻿​​‌‌​​​‌​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​​‌​​​‌‌​​‌​​​‌‌​​​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌​​‌​​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​‌​​​​‌‌​‌‌‌​​‌‌​​‌​​​‌‌​‌​​​​‌‌​​‌‌​​‌‌​​​​​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​​‌‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​​‌‌​​‌‌​‌​‌​​‌‌‌​​​​​‌‌‌​​‌​​‌‌​‌​‌​​‌‌​‌​‌​​‌‌‌​​‌​​‌‌‌​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​​​‌​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​‌‌​​​‌‌​​‌​​​‌‌​‌‌​​​‌‌​​​‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​‌‌‌​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​‌​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌​​​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​‌​​​‌‌​​​​​​‌‌​​‌‌​​‌‌​‌‌‌​​‌‌​‌​‌​​‌‌​‌​​​​‌‌​​​​​​‌‌‌​​‌​​‌‌​‌‌​​​‌‌​‌‌​​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​‌​​​​‌‌‌​​‌​​‌‌​​‌‌​​‌‌‌​​​​​‌‌​​​‌​​‌‌​‌​​​​‌‌‌​​​​​‌‌​​‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​​​​‌‌​​‌‌​​‌‌​‌​‌​​‌‌‌​​​​​‌‌​​‌​​​‌‌​​​​​​‌‌‌​​‌​​‌‌​​​‌​​‌‌‌​​‌​​‌​​​​​​​‌‌​‌‌‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌‌​​​​​‌‌​​​​​​‌‌​‌‌‌​​‌‌​‌​‌​​‌‌​‌​​​​‌‌​‌​‌​​‌​​​​​​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​​​​​​‌‌​‌​‌​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌​​​​‌‌​‌​​​​‌‌‌​​‌​​‌‌‌​​​​​‌‌​​​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌​​‌‌​​‌‌​​​​​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​​‌‌​​‌‌​​‌​​​‌‌‌​​‌​​‌‌‌​​‌​​‌‌‌​​​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​​‌​​​‌​​​​​​​‌‌​​​‌​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​‌‌​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​‌‌​​​‌‌​​​‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​​‌‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​​​​​​‌‌​‌​‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​​​​​​‌‌​‌‌​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌​​‌‌​​‌‌​‌​​​​‌‌​‌‌​​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​​​​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​​‌‌​​‌‌‌​​​​​‌‌​‌​‌​​‌​​​​​​​‌‌‌​​​​​‌‌​‌‌​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​‌‌‌​​‌‌‌​​‌​​‌‌​‌​​​​‌‌​‌​‌​​‌‌​‌‌‌​​‌​​​​​​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌​‌​​‌‌​‌‌​​​‌‌‌​​‌​​‌‌​​​​​​‌‌​‌​​​​‌‌‌​​​​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​​‌‌​​‌‌​‌​​​​‌‌​​​​​​‌‌​​‌​​​‌‌‌​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​​​​​​‌‌‌​​‌​​‌‌​​‌‌​​‌‌​​​​​​‌‌​​​‌​​‌​​​​​​​‌‌‌​​​​​‌‌‌​​‌​​‌‌​‌‌​​​‌‌​​​​​​‌‌​​‌‌​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​​​​​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​​‌‌​​‌‌​‌​​​​‌‌​‌‌‌​​‌‌​‌​​​​‌‌​​‌‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌​​​​​​​‌‌​‌‌​​​‌‌​‌​​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​​​​​​‌‌​​​​​​‌‌​‌‌​​​‌‌​​​​​​‌‌​‌​‌​​‌​​​​​​​‌‌​​‌​​​‌‌​​​​​​‌‌​‌​‌​​‌‌​​‌‌​​‌‌‌​​‌​​‌‌​‌‌‌​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌‌​​​​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​​​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌​​‌​​​‌‌​‌‌​​​‌‌​‌​​﻿

valid_input = [1,2]
while input:
    try:
        input = int(input("give me your input sweetie:"))
        if input in valid_input:
            break
    except ValueError:
        print("❌❌❌ i need a number dumbass, 1 means decode and 2 means encode")
        continue
    except:
        print("❌❌❌ i need a damn 1 or 2")
        break

pre_m = """something cryptic encrypted in binary"""
m_binary = ''.join(format(ord(char), '08b') for char in pre_m)
m = list(m_binary)
mm = 0
pre_n = """some more text to be encrypted"""
n_binary = ''.join(format(ord(char), '08b') for char in pre_n).replace("1","2").replace("0","1")
n = list(n_binary)
nn = 0

write_file = open("output.txt","wt")
read_file  = open("input.txt","rt")
m_chars = 0
n_chars = 0
file_m_binary = ""
file_n_binary = ""

for line in read_file:
    m_chars = line.strip()[12:13:]
    file_m_binary += str(m_chars)
    n_chars = line.strip()[15:16:]
    file_n_binary += str(n_chars)

times = len(chars) * len(chars) * len(chars)

q = 0 
w = 0
e = 0 #i think one letter one are fine for that
    
if input == 1:
    for something in range(int(input)):
        #pre_m_binary = int(file_m_binary, 2)
        #pr__n_binary = file_n_binary.replace("1","0").replace("2","1")
        #pre_n_binary = int(pr__n_binary, 2)
        #fixed_m_binary = pre_m_binary.to_bytes((pre_m_binary.bit_length() + 7) // 8, 'big').decode(encoding="utf-8",errors="ignore")
        #fixed_n_binary = pre_n_binary.to_bytes((pre_n_binary.bit_length() + 7) // 8, 'big').decode(encoding="utf-8",errors="ignore")
        #print(fixed_m_binary,file=write_file)
        #print("decoded stuff №1 is: ",fixed_m_binary)
        #print("raw binary of the №1:",file_m_binary)
        print(file_m_binary,file=write_file)
        #print(fixed_n_binary,file=write_file)
        #print("decoded stuff №2 is: ",fixed_n_binary)
        #print("raw binary of the №2:",file_n_binary.replace("1","0").replace("2","1"))
        print(file_n_binary.replace("1","0").replace("2","1"),file=write_file)
        print("✅✅✅ take the binary from the output file and decode them on https://www.rapidtables.com/convert/number/binary-to-ascii.html , i cant do that reliably in python")
        print("take that binary and decode them on https://www.rapidtables.com/convert/number/binary-to-ascii.html , i cant do that reliably in python",file=write_file)
        write_file.close()
        read_file.close()

if input == 2:
    for something in range(times):
        print("random roll ",end="",file=write_file)
        print(m[mm] if mm < len(m) else "0",end="",file=write_file)
        print("..",end="",file=write_file)
        print(n[nn] if nn < len(n) else "1",end="",file=write_file)
        print(" sbw:",end="",file=write_file)
        print(chars[e]+chars[w]+chars[q],file=write_file)
        mm += 1
        nn += 1
        if mm == len(m):
            mm = len(m)
        if nn == len(n):
            nn = len(n)
        q += 1
        if q == len(chars):
            q = 0
            w += 1
        if w == len(chars):
            w = 0
            e += 1
    print("✅✅✅ should be ready, idunno for sure")
    write_file.close()
    read_file.close()

File: text_generator/local_text_generator.lua
local chars = {"0","1","2","3","4","5","6","7","8","9","a","b","c","d","e","f","g","h","i"}--what characters code can use and in what order or idunno
--     ^^^ alias bytes ﻿​​‌‌​​​‌​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​​‌​​​‌‌​​‌​​​‌‌​​​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌​​‌​​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​‌​​​​‌‌​‌‌‌​​‌‌​​‌​​​‌‌​‌​​​​‌‌​​‌‌​​‌‌​​​​​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​​‌‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​​‌‌​​‌‌​‌​‌​​‌‌‌​​​​​‌‌‌​​‌​​‌‌​‌​‌​​‌‌​‌​‌​​‌‌‌​​‌​​‌‌‌​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​​​‌​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​‌‌​​​‌‌​​‌​​​‌‌​‌‌​​​‌‌​​​‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​‌‌‌​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​‌​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌​​​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​‌​​​‌‌​​​​​​‌‌​​‌‌​​‌‌​‌‌‌​​‌‌​‌​‌​​‌‌​‌​​​​‌‌​​​​​​‌‌‌​​‌​​‌‌​‌‌​​​‌‌​‌‌​​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​‌​​​​‌‌‌​​‌​​‌‌​​‌‌​​‌‌‌​​​​​‌‌​​​‌​​‌‌​‌​​​​‌‌‌​​​​​‌‌​​‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​​​​‌‌​​‌‌​​‌‌​‌​‌​​‌‌‌​​​​​‌‌​​‌​​​‌‌​​​​​​‌‌‌​​‌​​‌‌​​​‌​​‌‌‌​​‌​​‌​​​​​​​‌‌​‌‌‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌‌​​​​​‌‌​​​​​​‌‌​‌‌‌​​‌‌​‌​‌​​‌‌​‌​​​​‌‌​‌​‌​​‌​​​​​​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​​​​​​‌‌​‌​‌​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌​​​​‌‌​‌​​​​‌‌‌​​‌​​‌‌‌​​​​​‌‌​​​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌​​‌‌​​‌‌​​​​​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​​‌‌​​‌‌​​‌​​​‌‌‌​​‌​​‌‌‌​​‌​​‌‌‌​​​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​​‌​​​‌​​​​​​​‌‌​​​‌​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​‌‌​​​‌‌​​​‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​‌‌​​​‌‌​​​‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​​‌‌​​‌‌​​‌​​​‌‌​​‌‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​​​​​​‌‌​‌​‌​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​​​​​​‌‌​‌‌​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌​​‌‌​​‌‌​‌​​​​‌‌​‌‌​​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​​​​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​​‌‌​​‌‌‌​​​​​‌‌​‌​‌​​‌​​​​​​​‌‌‌​​​​​‌‌​‌‌​​​‌‌​​‌​​​‌‌​‌​​​​‌‌​‌‌‌​​‌‌‌​​‌​​‌‌​‌​​​​‌‌​‌​‌​​‌‌​‌‌‌​​‌​​​​​​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌​‌​​‌‌​‌‌​​​‌‌‌​​‌​​‌‌​​​​​​‌‌​‌​​​​‌‌‌​​​​​‌‌​​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌​​​​‌‌​​‌‌​​‌‌​‌​​​​‌‌​​​​​​‌‌​​‌​​​‌‌‌​​‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌‌​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​​​​​​‌‌‌​​‌​​‌‌​​‌‌​​‌‌​​​​​​‌‌​​​‌​​‌​​​​​​​‌‌‌​​​​​‌‌‌​​‌​​‌‌​‌‌​​​‌‌​​​​​​‌‌​​‌‌​​‌‌‌​​​​​‌‌​‌​‌​​‌‌​​​​​​‌‌​‌‌‌​​‌​​​​​​​‌‌​​​‌​​‌‌​‌‌​​​‌‌​​‌‌​​‌‌​‌​​​​‌‌​‌‌‌​​‌‌​‌​​​​‌‌​​‌‌​​‌‌‌​​‌​​‌‌​​‌​​​‌‌​​‌‌​​‌​​​​​​​‌‌​‌‌​​​‌‌​‌​​​​‌‌​‌​‌​​‌‌​​​‌​​‌‌​​​​​​‌‌​​​​​​‌‌​‌‌​​​‌‌​​​​​​‌‌​‌​‌​​‌​​​​​​​‌‌​​‌​​​‌‌​​​​​​‌‌​‌​‌​​‌‌​​‌‌​​‌‌‌​​‌​​‌‌​‌‌‌​​‌‌‌​​​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌‌​​​​​‌​​​​​​​‌‌​​​‌​​‌‌‌​​​​​‌‌​‌‌​​​‌‌​‌​‌​​‌‌​‌​​​​‌‌​‌​​​​‌‌​‌‌​​​‌‌​​‌​​​‌‌​‌‌​​​‌‌​‌​​﻿
print("give me your input sweetie") --say that the programs is waiting for an input
local input = tonumber(io.read())

local times = tonumber(#chars * #chars * #chars) -- how many times code needs to run; using hashtag before a variable makes the output be its length, instead of just copying the value

io.output("output.txt")
io.input ("input.txt" ) --makes so that the default reading and writing is controlled between them
local file = io.read("a")

local binaryconversion = nil --thats just is required for function, ask the moon why its like that

-- thats is some code from gemini irc so uhm
function binaryconversion(str)
    local binaryResult = ""
    for i = 1, #str do
        local byte = string.byte(str, i) -- Get ASCII/byte value
        local bits = ""
        for j = 7, 0, -1 do
            -- Extract each bit (8 bits per character)
            local bit = (byte >> j) & 1  -- Requires Lua 5.3+
            bits = bits .. bit
        end
        binaryResult = binaryResult .. bits -- Space added for readability
    end
    return binaryResult
end
-- i mean it works so ...
-- i just dont want to lie and not saying it feels wrong 

local array  = ""
local array2 = "" --initialize variables cuz lua said so
for i in string.gmatch(file, "%s(%d)%p%p%d%s") do -- select very specific numbers and dont think too much about 
    array = array..i
end
for i in string.gmatch(file, "%s%d%p%p(%d)%s") do
    array2 = array2..i --append what have been found
end

local byte2bin = nil
function byte2bin(n) -- credits goes to https://love2d.org/forums/viewtopic.php?t=84988&start=10 , im too dumb for ts
    local t, d = {}, 0
    d = math.log(n)/math.log(2) -- binary logarithm
    for i=math.floor(d+1),0,-1 do
        t[#t+1] = math.floor(n / 2^i)
        n = n % 2^i
    end
    return table.concat(t)
end

local pre_m = [[some text here]] --what is going to become a binary
local m_binary = binaryconversion(pre_m)
local m = {} or "0"
--m_binary:gsub(".", function(c) table.insert(m, c) end) --converting string into array/table/list, the {}
local mm = 1

local pre_n = [[mnBvcxzLKJHGFDsaPoIu]]
local n_binary = string.gsub(string.gsub(binaryconversion(pre_n), 1, 2), 0, 1) --convert the string into binary and make it so numbers are one up
local n = {} or "0"
local nn = 1

local file_m_binary =                         array
local file_n_binary = string.gsub(string.gsub(array2, 1, 0), 2, 1)

local q = 0
local w = 1
local e = 1 --just how many characters of code needs to paste, i think one letter one are fine for that

if input == 1 then
    io.write(file_m_binary.."\n"..file_n_binary.."\ntake that binary and decode them on https://www.rapidtables.com/convert/number/binary-to-ascii.html , i dont want to do that in lua just cuz")
    print("take the binary from the output file and decode them on https://www.rapidtables.com/convert/number/binary-to-ascii.html , i dont want to do that in lua just cuz")
elseif input == 2 then
    for i=1,times,1 do
        mm = mm+1
        nn = nn+1
        if mm == #m then
            mm = #m
        end
        if nn == #n then
            nn = #n
        end
        q = q+1
        if q == #chars+1 then
            q = 1
            w = w+1
        end
        if w == #chars+1 then
            w = 1
            e = e+1
        end
        m_binary:gsub(".", function(c) table.insert(m, c) end)
        n_binary:gsub(".", function(c) table.insert(n, c) end)
        io.write("random roll "..m[mm]..".."..n[nn].." sbw:"..chars[e]..chars[w]..chars[q].."\n")
    end
    print("should be ready, idunno for sure")
elseif input == 3 then
    local count = 1
    local byte = tostring(string.byte(pre_m,count) or 0)
    local bite = {}
    byte:gsub(".", function(c) table.insert(bite, c) end)
    for i = 1, #pre_m, 1 do
        if count <= #byte+1 or count == #byte+1 then
            io.write(byte2bin(bite[count]))
            count = count+1
        else
            byte:gsub(".", function(c) table.insert(bite, c) end)
            count = 1
        end
    end
else
    print("i need either 1 or 2 dumbass, 1 means decode and 2 means encode")
end

